feat: allow resources to be created without namespace defined

danielhelfand opened the pull request:
Add a step to kubedog to allow resources to be created without defining a namespace in manifests/tempaltes.

Signed-off-by: Daniel Helfand <helfand.4@gmail.com>

danielhelfand commented on kubedog.go:
```diff
@@ -50,6 +50,7 @@ func (kdt *Test) Run() {
 	kdt.scenarioContext.Step(`^a Kubernetes cluster$`, kdt.KubeContext.AKubernetesCluster)
 	kdt.scenarioContext.Step(`^(?:I )?(create|submit|delete) (?:the )?resource ([^"]*)$`, kdt.KubeContext.ResourceOperation)
 	kdt.scenarioContext.Step(`^(?:I )?(create|submit|delete) (?:the )?resources in ([^"]*)$`, kdt.KubeContext.MultiResourceOperation)
+	kdt.scenarioContext.Step(`^(?:I )?(create|submit|delete) (?:the )?resources in ([^"]*) in (?:the )?([^"]*) namespace`, kdt.KubeContext.MultiResourceOperationInNamespace)
```
Would appreciate some attention here. Not sure if step syntax makes sense here.

garomonegro replied:
I think you need the `$` at the end, not sure tho

danielhelfand replied:
I only see that for `([^"]*)`. Since it ends with "namespace", I think it might be okay without `$`.

danielhelfand replied:
Tested out locally and seems Gherkin was getting mapped to correct step/func.

garomonegro commented:
Looks good, thanks you @danielhelfand!

Now that we have `unstructuredResourceOperation` take a namespace, would you mind adding another step for single resources operations in given namespace? the function would be similar to `ResourceOperation` and the step definition would be something like:
```
kdt.scenarioContext.Step(`^(?:I )?(create|submit|delete) (?:the )?resource ([^"]*) in (?:the )?([^"]*) namespace$`, kdt.KubeContext.ResourceOperationInNamespace)
```

Also please add the newly added steps to [the docs](https://github.com/keikoproj/kubedog/blob/master/docs/syntax.md).

danielhelfand commented:
> Now that we have unstructuredResourceOperation take a namespace, would you mind adding another step for single resources operations in given namespace? the function would be similar to ResourceOperation and the step definition would be something like:

Added in https://github.com/keikoproj/kubedog/pull/62/commits/f7b3911b5473cc3ca3bb1d4b1da4b4693b09a287

> Also please add the newly added steps to [the docs](https://github.com/keikoproj/kubedog/blob/master/docs/syntax.md).

Added here: https://github.com/keikoproj/kubedog/pull/62/commits/8dfd8fe72b9242552d6141e89b05359d4be90782
